airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -620,10 +629,10 @@ def create_legacy_to_per_partition_state_migration(
             )
 
         return LegacyToPerPartitionStateMigration(
-            declarative_stream.retriever.partition_router,
-            declarative_stream.incremental_sync,
+            partition_router,
+            declarative_stream.incremental_sync,  # type: ignore # was already checked. Migration can be applied only to incremental streams.
```

Comment:
Where was this checked? I only see if statements on `declarative_stream.retriever.partition_router`

Reply:
added check for this